plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshViewImpl.java

```diff
@@ -0,0 +1,344 @@
+            @Override
+            public void onKeyUp(KeyUpEvent keyUpEvent) {
+                delegate.onUserNameChanged(userName.getValue());
+            }
+        });
+
+        password.addKeyUpHandler(new KeyUpHandler() {
+            @Override
+            public void onKeyUp(KeyUpEvent keyUpEvent) {
+                delegate.onPasswordChanged(password.getValue());
+            }
+        });
+
+        rootElement.setVisible(true);
+    }
+
+    private Button createButton(String title, String debugId, ClickHandler clickHandler) {
+        Button button = new Button();
+        button.setText(title);
+        button.ensureDebugId(debugId);
+        button.getElement().setId(debugId);
+        button.addStyleName(this.windowResources.windowCss().button());
+        button.addClickHandler(clickHandler);
+        //set default tab index
+        button.setTabIndex(0);
+        return button;
+    }
+
+
+    @Override
+    public Widget asWidget() {
+        return rootElement;
+    }
+
+
+    @Override
+    public void setDelegate(ActionDelegate delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public void setTargetName(String targetName) {
+        this.targetName.setValue(targetName);
+    }
+
+    @Override
+    public String getTargetName() {
+        return targetName.getValue();
+    }
+
+    @Override
+    public void markTargetNameInvalid() {
+        targetName.markInvalid();
+    }
+
+    @Override
+    public void unmarkTargetName() {
+        targetName.unmark();
+    }
+
+    @Override
+    public void markHostInvalid() {
+        host.markInvalid();
+    }
+
+    @Override
+    public void unmarkHost() {
+        host.unmark();
+    }
+
+    @Override
+    public void markPortInvalid() {
+        port.markInvalid();
+    }
+
+    @Override
+    public void unmarkPort() {
+        port.unmark();
+    }
+
+    @Override
+    public void setHost(String host) {
+        this.host.setValue(host);
+    }
+
+    @Override
+    public String getHost() {
+        return host.getValue();
+    }
+
+    @Override
+    public void setPort(String port) {
+        this.port.setValue(port);
+    }
+
+    @Override
+    public String getPort() {
+        return port.getValue();
+    }
+
+    @Override
+    public void setUserName(String userName) {
+        this.userName.setValue(userName);
+    }
+
+    @Override
+    public String getUserName() {
+        return userName.getValue();
+    }
+
+    @Override
+    public void setPassword(String password) {
+        this.password.setValue(password);
+    }
+
+    @Override
+    public String getPassword() {
+        return password.getValue();
+    }
+
+    @Override
+    public void enableSaveButton(boolean enable) {
+        saveButton.setEnabled(enable);
+    }
+
+    @Override
+    public void enableCancelButton(boolean enable) {
+        cancelButton.setEnabled(enable);
+    }
+
+    @Override
+    public boolean restoreTargetFields(SshMachineTarget target) {
```

Comment:
Please, justify. Why do you want to use Button widgets directly in presenter(not setters methods from view)?